ui/pages/confirm-import-token/confirm-import-token.js

```diff
@@ -0,0 +1,134 @@
+import React, { useContext, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import {
+  ASSET_ROUTE,
+  IMPORT_TOKEN_ROUTE,
+} from '../../helpers/constants/routes';
+import Button from '../../components/ui/button';
+import Identicon from '../../components/ui/identicon';
+import TokenBalance from '../../components/ui/token-balance';
+import { I18nContext } from '../../contexts/i18n';
+import { MetaMetricsContext } from '../../contexts/metametrics';
+import { getMostRecentOverviewPage } from '../../ducks/history/history';
+import { addTokens, clearPendingTokens } from '../../store/actions';
+
+const getTokenName = (name, symbol) => {
+  return typeof name === 'undefined' ? symbol : `${name} (${symbol})`;
+};
+
+const ConfirmImportToken = () => {
+  const metricsEvent = useContext(MetaMetricsContext);
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const mostRecentOverviewPage = useSelector((state) =>
+    getMostRecentOverviewPage(state),
+  );
+  const pendingTokens = useSelector((state) => state.metamask.pendingTokens);
+
+  const tokenAddedEvent = (pendingToken) => {
+    metricsEvent({
```

Comment:
I think this action you're using isn't equivalent to the one that was being used before and this one requires different arguments. That's why the e2e test is failing — because this fails before `history.push` is called. You'll want to take a look at other places that use `metricsEvent` to know what to pass. I think it may be something like

``` javascript
metricsEvent({
  eventOpts: {
    name: 'Token Added',
    category: 'Wallet',
  }
})
```

I am not sure that `sensitiveProperties` is being used in this case so I am not sure it is necessary. But then again I am not completely familiar with how the metrics stuff works. Perhaps someone else can chime in here.

Reply:
So the eventOpts parameter is for the old metametrics.The new one, which the old implementation used too, has the payload that is mentioned here. So what is in the code is correct but the wrong context is being used. This is because we don't export the context from the new provider to prevent it from being used. Instead, using the useNewMetricEvent hook from the 'useMetricEvent'  file should be used. 

The sensitiveProperties here is used to keep on chain data anonymous in our metrics payload. Technically source, and unlisted should be hoisted into the `properties` payload. If we tracked the others to a metrics id then we could, in theory, identify a wallet -> metametrics id by examining the tokens for which wallets have balances and aligning them to the ids we know have those tokens in their metamask. By specifying in sensitiveProperties we make this orders of mangitude more difficult (preserving user privacy)

Reply:
@mcmire @brad-decker 

I believe I understand. I initially tried to maintain logic by preserving the context used by the previous code, and to do this I used this `metricsEvent` here. Thank you two for all of this info. Glad to know we can transition to `useNewMetricEvent` with the same args from our older `metricsEvent` method and to get this clarification with `sensitiveProperties`.

`useNewMetricEvent` was a bit tricky to use with dynamic props since it returns a `useCallback` hook. I consulted with @ryanml a bit about what we should do in this case and we thought utilizing `useState` might be fine here. Then I updated the code. Please let me know what y'all think.

During these updates, I found the `history.push` issue. My mistake was replacing `componentDidMount` with `useEffect` and forgetting to force this hook to run just once by passing an empty dependency list. This should be fixed now.

Reply:
@mcmire @digiwand `metricsEvent` is the context method name for the old analytics system context that was converted to translate the old metrics payloads into ones that work with the new ones. those payloads have the `eventOpts` object definition in them like @mcmire pointed out. 

`trackEvent` is the context value that the old code used, which is the correct new method for the new metametrics context. Yes its confusing.

in terms of the useCallback problem, this has been expressed before. As long as your current implementation is working and sending the right event payload then this is fine. I'll take into consideration how we might make the context better for consumption directly. To be fair, your original implementation using context would have been fine if it was the NEW metametrics context and the 'trackEvent' method. 😬 sorry about that.

Reply:
@brad-decker @mcmire Ooh like using 
```
import { MetaMetricsContext } from '../../contexts/metametrics.new';
```
instead of 
```
import { MetaMetricsContext } from '../../contexts/metametrics';
```

I was trying to preserve the context, but maybe I misunderstood the context used in the older, class implementation. I don't fully understand the differences between each of the providers, but I think I understand we should stick to using this newer context; ideally using `useNewMetricEvent` or possibly `import { MetaMetricsContext } from '../../contexts/metametrics.new';` depending on the circumstances. Thanks for elaborating on this!

Reply:
No problem. With the class based approach its literally just the *name* of the context item. If you review the two metrics contexts one is trackEvent the other is metricsEvent or something ... which name you used indicates which context you get lol.